great_expectations/datasource/new_datasource.py

```diff
@@ -292,6 +293,11 @@ def _validate_batch_request(self, batch_request: BatchRequest):
                 match Datasource name: "{self.name}".
                 """
             )
+        if not (batch_request.data_connector_name in self.data_connectors.keys()):
```

Comment:
Again, this feels to be not as good of a place for validating the attributes of `BatchRequest` as in the current version.  I may be missing a point here throughout, so happy to discuss.  Thanks!